server/collectionPub/api.ts

```diff
@@ -57,7 +57,8 @@ export const collectionPubServer = s.router(contract.collectionPub, {
 		});
 		return {
 			status: 201,
-			body: collectionPub,
+			// FIXME: Do some checking of whether the contextHint is valid
+			body: collectionPub as any,
```

Comment:
I tried to make collectionPubs and Collections more strict, by matching the types of contextHints and shape of the metadata with the Kind of the Collection, e.g. Issue collections can't have ISBNs, its pubs can only have Article contextHints etc. 
This turned out to be quite difficult type wise, especially because collection kind information is not available when updating collectionPubs or Collections (you cannot change a collection's kind afaik, or should not be able to at least).

While I do think it would be good to not allow people to e.g. set ISBNs on Issues, this would require some rewriting of the routes (adding extra checks for kinds). I might try to add this to the routes PR, but atm it's not very important.